algorithm_toolkit/views/home.py

```diff
@@ -131,106 +132,123 @@ def show_docs(filename):
+                save_path = os.path.join(path, 'history', save_fname)
+            c_obj.chain_ledger.save_history_to_json(save_path, pretty=True)
+
+            if 'CHAIN_HISTORY' in app.config:
+                if app.config['CHAIN_HISTORY_LENGTH'] > 0:
+                    ch = app.config['CHAIN_HISTORY']
+
+                    if len(ch) > app.config['CHAIN_HISTORY_LENGTH']:
+                        ch.popitem(last=False)
+
+                    ch[status_key] = c_obj.chain_ledger
         else:
-            run_mode = 'single'
-    elif request.method == 'GET':  # pragma: no branch
-        chain = request.args.get('chain', None)
-        status_key = request.args.get('status_key', None)
-        run_mode = request.args.get('run_mode', 'single')
-        iter_param = request.args.get('iter_param', None)
-        iter_type = request.args.get('iter_type', None)
-        iter_value = request.args.get('iter_value', None)
-
-    if status_key is None:
-        status_key = create_random_string(http_safe=True)
-
-    if chain is None:
-        return make_response('Missing chain parameter in request', 400)
-
-    if chain.lower() == 'from_global':
-        try:
-            chain = app.config['CHAIN_DATA'].pop(status_key)
-        except ValueError:
-            return make_response(
-                'Chain parameter not present in app configuration', 400)
-    else:
-        try:
-            chain = json.loads(chain)
-        except ValueError:
-            return make_response('Chain parameter not properly formatted', 400)
+            response = c_obj.call_batch(iter_param, iter_type, iter_value)
+
+        cl.remove_working_folders()
 
-    if 'chain_name' not in chain:
-        return make_response('Chain name not defined', 400)
+        if response['output_type'] == 'error':
+            return make_response(jsonify(response), 400)
 
-    if 'algorithms' not in chain:
-        return make_response('Algorithms not defined', 400)
+    else:  # If check not a dict, then it must be a bad request!
+        return check
 
-    c_obj = AlgorithmChain(path, chain)
-    if c_obj.chain_definition == {}:
-        return make_response('Chain name not found', 404)
-    cl = c_obj.create_ledger(status_key)
-    cl.make_working_folders()
+    return jsonify(response)
 
-    if run_mode == 'single':
-        response = c_obj.call_chain_algorithms()
-        save_fname = status_key + '.json'
 
-        if 'CHAIN_LEDGER_HISTORY_PATH' in app.config:
-            save_path = os.path.join(
-                app.config['CHAIN_LEDGER_HISTORY_PATH'], save_fname)
-        else:
-            make_dir_if_not_exists(os.path.join(path, 'history'))
-            save_path = os.path.join(path, 'history', save_fname)
-        c_obj.chain_ledger.save_history_to_json(save_path, pretty=True)
+# This route/function is depricated ( Replaced by api_chain() )
+# Chain name expected in the request form
+@home.route('/main/', methods=['POST', 'GET'])
```

Comment:
I'm guessing there is common code between this endpoint and the new endpoint above? If so maybe we can refactor out all the common pieces into a single method. This would give us a start to a standard python library/api.

Reply:
There is. That is the main reason I split some of the code into utils. I should have also put the remaining pieces in another method. We can maybe chat about this tomorrow (: